packages/chain-adapters/src/types.ts

```diff
@@ -83,13 +83,15 @@ export type FeeDataEstimate<T extends ChainId> = {
 
 export type SubscribeTxsInput = {
   wallet: HDWallet
-  bip44Params: BIP44Params
+  accountNumber: number
```

Comment:
the core of this PR, everything else hangs off this

now that we only *require* `accountNumber`, with optional `index` and `isChange` params (for UTXO chains) below, we use the `getBIP44Params` utility function of the respective base chain adapter to generate the correct `BIP44Params`, derived from the safe defaults defined on the concrete implementation of the adapter, plus whatever we pass in, which is also validated.

this prevents us from passing consuming clients (web) from passing non-standard `BIP44Params`, specifically `purpose` and `coinType` into a chain adapter, and generating non-standard addresses, leading to non-trivially recoverable funds